### PR TITLE
Take a stab at error handling.

### PR DESCRIPTION
**Pull request by tikue**
I'm testing this out locally to get connections to drop, so that I can use a default global reactor and not have tests hang.

I doubt this is how it should really be done, but I figure it might be better than panicking, at least. ;) Mostly I just wanted to get a conversation started about error handling!

### CODE REVIEW

File: src/proto/pipeline/client.rs [hunk 1]
```diff
@@ -104,10 +104,17 @@ impl<T, E> Task for Client<T, E>
                             let c = self.in_flight.remove(0);
                             c.complete(resp);
                         }
+                        Frame::Error(e) => {
+                            if !self.in_flight.is_empty() {
+                                let c = self.in_flight.remove(0);
```
Comment (tikue) [inline src/proto/pipeline/client.rs]: Should `in_flight` be a `VecDeque`? Popping from the front is O(n) for `Vec`s.

Comment (arthurprs) [inline src/proto/pipeline/client.rs]: Ya, I came to say the same.

Comment (carllerche) [inline src/proto/pipeline/client.rs]: It basically was a `Vec` to get something done. It should definitely not stay a `Vec`

File: src/proto/pipeline/client.rs [hunk 2]
```diff
@@ -84,7 +84,7 @@ impl<T, U, E> Clone for ClientHandle<T, U, E>
 }
 
 impl<T, E> Task for Client<T, E>
-    where T: Transport,
+    where T: Transport<Error = E>,
```
Comment (tikue) [inline src/proto/pipeline/client.rs]: Is this correct?

**Comment by carllerche** [comment]
Hey, thanks for the PR. I implemented error handling in part based from this PR, so it was quite helpful :)

**Pull request closed without merging**